File: LetsPlant/app/src/main/java/com/e/letsplant/activities/SplashScreen.java
package com.e.letsplant.activities;

import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.NonNull;

import com.google.firebase.auth.FirebaseAuth;

public class SplashScreen extends MainActivity{

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

    }

    private final FirebaseAuth.AuthStateListener fAuthListener = new FirebaseAuth.AuthStateListener() {
        @Override
        public void onAuthStateChanged(@NonNull FirebaseAuth firebaseAuth) {
            Intent intent;
            String userUid = firebaseAuth.getUid();
            if (userUid != null) {
                intent = new Intent(getApplicationContext(), SecondActivity.class);
            } else {
                intent = new Intent(getApplicationContext(), SignActivity.class);
            }
            startActivity(intent);
            finish();
        }
    };

    @Override
    public void onStart() {
        super.onStart();
        firebaseAuth.addAuthStateListener(fAuthListener);
    }

    @Override
    public void onStop() {
        super.onStop();
        if (fAuthListener != null) {
            firebaseAuth.removeAuthStateListener(fAuthListener);
        }
    }
}

File: LetsPlant/app/src/main/java/com/e/letsplant/activities/SplashScreen.kt
package com.e.letsplant.activities

import android.content.Intent
import com.google.firebase.auth.FirebaseAuth.AuthStateListener

class SplashScreen : MainActivity() {

    private val fAuthListener: AuthStateListener =
        AuthStateListener { firebaseAuth ->
            val intent: Intent
            val userUid: String? = firebaseAuth.uid
            if (userUid != null) {
                intent = Intent(applicationContext, SecondActivity::class.java)
            } else {
                intent = Intent(applicationContext, SignActivity::class.java)
            }
            startActivity(intent)
            finish()
        }

    public override fun onStart() {
        super.onStart()
        firebaseAuth!!.addAuthStateListener((fAuthListener))
    }

    public override fun onStop() {
        super.onStop()
        firebaseAuth!!.removeAuthStateListener(fAuthListener)
    }
}